src/agent/config.ts

```diff
@@ -17,7 +17,7 @@
 import * as common from '../types/common';
 
 export interface DebugAgentConfig extends common.AuthenticationConfig {
-  workingDirectory?: string;
+  workingDirectory: string;
```

Comment:
This shouldn't block landing, but I feel like any user-facing config structure should have all properties optional. Otherwise if I only wanted to specify `logLevel`, it looks like I'm required `workingDirectory` as well. If it's addressed it should be in a different PR.

Reply:
Do we now making workingDirectory a required config now? I would support this decision as debug agent will not be responsible to find the correct script path. That is also helpful that we will not have ambiguous path problem as we will just using the workingDirectory + filename as the complete path for the file.

Reply:
I agree.  I think we need a `ResolvedDebugAgentConfig` as well as a `DebugAgentConfig`.  The `DebugAgentConfig` is the config provided by the user and has all optional fields.  A `ResolvedDebugAgentConfig` is based on the user provided config but uses default values for values not provided by the user.

Then the rest of the code will use a `ResolvedDebugAgentConfig`.  Thus, from the code's perspective, some options are not optional, but from the user's perspective, all options are optional.